plugin/kubernetes/pvc.go

```diff
@@ -181,28 +199,122 @@ func (v *pvc) runInTemporaryPod(ctx context.Context, cmd []string) ([]byte, erro
 	return bytes, nil
 }
 
+func (v *pvc) runInExistingPod(ctx context.Context, mount mountInfo, cmder makeCommandAtPath) ([]byte, error) {
+	cmd := cmder(mount.path)
+	activity.Record(ctx, "Executing in existing pod: %v", cmd)
+	execRequest := v.client.CoreV1().RESTClient().Post().
+		Resource("pods").
+		Name(mount.pod).
+		Namespace(v.namespace).
+		SubResource("exec").
+		Param("command", cmd[0]).
+		Param("container", mount.container).
+		Param("stderr", "true").
+		Param("stdout", "true")
+
+	for _, arg := range cmd[1:] {
+		execRequest = execRequest.Param("command", arg)
+	}
+
+	executor, err := remotecommand.NewSPDYExecutor(v.config, "POST", execRequest.URL())
+	if err != nil {
+		return nil, err
+	}
+
+	var stdout, stderr bytes.Buffer
+	streamOpts := remotecommand.StreamOptions{
+		Stdout: &stdout,
+		Stderr: &stderr,
+	}
+	err = executor.Stream(streamOpts)
+	activity.Record(ctx, "stdout: %v", stdout.String())
+	activity.Record(ctx, "stderr: %v", stderr.String())
+	if err != nil {
+		return nil, err
+	}
+	return stdout.Bytes(), nil
+}
+
+type mountInfo struct {
+	pod, container, path string
+}
+
+// TODO: return read-write mount if available (fallback to read-only) that mounts the volume
```

Comment:
Why read-write mount? Guessing so we can write to it? Also, does the latter half of the TODO ("try mounting in a new pod ...") belong here? That seems like something handled by `runInTemporaryPod`

Reply:
Read-write mount is so Write works (once it's added to Volume). I can split the comment up, but it felt all a single idea of optimizing more situations.